app-frontend/src/app/components/lab/templateItem/templateItem.module.js

```diff
@@ -10,10 +10,39 @@ const TemplateItemComponent = {
 };
 
 class TemplateItemController {
-    constructor() {
+    constructor($rootScope, $log, userService) {
         'ngInject';
+        $rootScope.autoInject(this, arguments);
         this.BUILDCONFIG = BUILDCONFIG;
     }
+
+    $onInit() {
+        this.getTemplateOwner();
+    }
+
+    getTemplateOwner() {
+        if (this.templateData.owner === 'default') {
+            this.templateOwner = this.BUILDCONFIG.APP_NAME;
+        } else {
+            this.userService.getUserById(this.templateData.owner).then(user => {
+                this.setTemplateOwnerDisplay(user);
+            }, err => {
+                this.$log.error(err);
+                this.templateOwner = this.BUILDCONFIG.APP_NAME;
+            });
+        }
+    }
+
+    setTemplateOwnerDisplay(user) {
+    // user.id is a desperate last resort
+        this.templateOwner = _.chain([
```

Comment:
There are a few things here that I think need addressing.

1. If first name if defined, we show only the first name, which seems odd. So I would change the first option to be a concatenation of both first and last name.

2. I think we should remove email from the list of options here since we are moving towards not ever exposing email addresses for privacy reasons.

3. We could accomplish the fallback values by using `||` instead of this series of method calls: `(user.personalInfo.firstName + ' ' + user.personalInfo.lastName).trim() || user.name || user.id` would achieve the same thing.